File: C++/0277.cpp
#include <iostream>
using namespace std;


int main()
{
    int a, b, c;
    bool isFirst = true;

    cin >> a >> b >> c;

    if (a != 0)
    {
        cout << a;
        isFirst = false;
    }
    if (b != 0)
    {
        if (b < -1)
            cout << b << "x";
        else if (b == -1)
            cout << "-x";
        else if (b > 1)
            cout << (isFirst ? "" : "+") << b << "x";
        else
            cout << (isFirst ? "" : "+") << "x";
        isFirst = false;
    }
    if (c != 0)
    {
        if (c < -1)
            cout << c << "y";
        else if (c == -1)
            cout << "-y";
        else if (c > 1)
            cout << (isFirst ? "" : "+") << c << "y";
        else
            cout << (isFirst ? "" : "+") << "y";
        isFirst = false;
    }
    if (isFirst)
        cout << "0";
    cout << endl;

    return 0;
}

File: Python/0277.py
a, b, c = map(int, input().split())
isFirst = True

if a != 0:
    print(a, end='')
    isFirst = False
if b != 0:
    if b < -1:
        print(b, 'x', sep='', end='')
    elif b == -1:
        print('-x', end='')
    else:
        if not isFirst:
            print('+', end='')
        if b != 1:
            print(b, end='')
        print('x', end='')
    isFirst = False
if c != 0:
    if c < -1:
        print(c, 'y', sep='', end='')
    elif c == -1:
        print('-y', end='')
    else:
        if not isFirst:
            print('+', end='')
        if c != 1:
            print(c, end='')
        print('y', end='')
    isFirst = False
if isFirst:
    print('0', end='')
print()
